bin/_convert_doc_templates_with_gpt

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+if [ -z "$API_KEY" ]; then
+  echo "No API Key present. Please provide one"
+  exit 1
+fi
+
+MODEL="gpt-4o-mini"
+DOCS_DIR="docs"
+
+# Used for generating links inside of the documentation
+get_repository_url() {
+    REMOTE_URL=$(git config --get remote.origin.url)
+
+    if [ -z "$REMOTE_URL" ]; then
+        echo "No remote origin found. Ensure that this is a valid Git repository."
+        return 1
+    fi
+
+    # Convert the SSH URL to HTTP if necessary
+    if [[ "$REMOTE_URL" == git@* ]]; then
+        # Convert git@github.com:user/repo.git to https://github.com/user/repo
+        REMOTE_URL="https://${REMOTE_URL#git@}"
+        REMOTE_URL="${REMOTE_URL/:/\/}"
+        REMOTE_URL="${REMOTE_URL%.git}"
+    elif [[ "$REMOTE_URL" == *.git ]]; then
+        # Remove .git suffix if URL is already in HTTP(S) format
+        REMOTE_URL="${REMOTE_URL%.git}"
+    fi
+
+    echo "${REMOTE_URL}/blob/main/"
+}
+
+REPOSITORY_BASE_URL=$(get_repository_url)
+
+call_gpt_api() {
+  local prompt="$1"
+
+  escaped_prompt=$(echo "$prompt" | jq -sR .)
+
+  response=$(curl -s -X POST https://api.openai.com/v1/chat/completions \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $API_KEY" \
+    -d "{
+      \"model\": \"$MODEL\",
+      \"messages\": [
```

Comment:
can you use heredoc, so that you don't have to escape the dblquotes?

Reply:
sure